Test.Microsoft.NET.Build.Containers.Filesystem/ParseContainerPropertiesTests.cs

```diff
@@ -11,108 +9,72 @@ public class ParseContainerPropertiesTests
-            task.ContainerRegistry = "http://localhost:5010";
-            task.ContainerImageName = "dotnet/testimage";
-            task.ContainerImageTag = "5.0";
-
-            Assert.IsTrue(task.Execute());
-            Assert.AreEqual("https://mcr.microsoft.com", task.ParsedContainerRegistry);
-            Assert.AreEqual("dotnet/runtime", task.ParsedContainerImage);
-            Assert.AreEqual("6.0", task.ParsedContainerTag);
-
-            Assert.AreEqual("dotnet/testimage", task.NewContainerImageName);
-            Assert.AreEqual("5.0", task.NewContainerTag);
-        }
-
-        [TestMethod]
-        public void BaseRegistriesWithNoSchemeGetHttps()
-        {
-            ParseContainerProperties task = new ParseContainerProperties();
-            task.FullyQualifiedBaseImageName = "mcr.microsoft.com/dotnet/runtime:6.0";
-            task.ContainerRegistry = "http://localhost:5010";
-            task.ContainerImageName = "dotnet/testimage";
-            task.ContainerImageTag = "5.0";
-
-            Assert.IsTrue(task.Execute());
-            Assert.AreEqual("https://mcr.microsoft.com", task.ParsedContainerRegistry);
-            Assert.AreEqual("dotnet/runtime", task.ParsedContainerImage);
-            Assert.AreEqual("6.0", task.ParsedContainerTag);
-
-            Assert.AreEqual("http://localhost:5010", task.NewContainerRegistry);
-            Assert.AreEqual("dotnet/testimage", task.NewContainerImageName);
-            Assert.AreEqual("5.0", task.NewContainerTag);
-        }
-
-        [TestMethod]
-        public void UserRegistriesWithNoSchemeGetHttps()
-        {
-            ParseContainerProperties task = new ParseContainerProperties();
-            task.FullyQualifiedBaseImageName = "mcr.microsoft.com/dotnet/runtime:6.0";
-            task.ContainerRegistry = "localhost:5010";
-            task.ContainerImageName = "dotnet/testimage";
-            task.ContainerImageTag = "5.0";
-
-            Assert.IsTrue(task.Execute());
-            Assert.AreEqual("https://mcr.microsoft.com", task.ParsedContainerRegistry);
-            Assert.AreEqual("dotnet/runtime", task.ParsedContainerImage);
-            Assert.AreEqual("6.0", task.ParsedContainerTag);
-
-            Assert.AreEqual("https://localhost:5010", task.NewContainerRegistry);
-            Assert.AreEqual("dotnet/testimage", task.NewContainerImageName);
-            Assert.AreEqual("5.0", task.NewContainerTag);
+            var (project, _) = Evaluator.InitProject(new () {
+                [ContainerBaseImage] = "mcr.microsoft.com/dotnet/runtime:6.0",
+                [ContainerRegistry] = "localhost:5010",
+                [ContainerImageName] = "dotnet/testimage",
+                [ContainerImageTag] = "5.0"
+            });
+            var instance = project.CreateProjectInstance(global::Microsoft.Build.Execution.ProjectInstanceSettings.None);
+            Assert.IsTrue(instance.Build(new[]{ComputeContainerConfig}, null, null, out var outputs));
```

Comment:
This pattern (make an instance and build a target) is fairly common, and you need a ProjectInstance if you want to get changed properties/items. Should I just commit to a helper that creates and returns a projectInstance instead? I also feel like we'll always want the `log` property so why not just assume that...

Reply:
sounds quite reasonable to me